internal/pkg/plugin/load.go

```diff
@@ -14,30 +14,15 @@ import (
 	pluginapi "github.com/sylabs/singularity/pkg/plugin"
 )
 
-// initialized stores whether or not the plugin system has been initialized. A
-// call to Initialize MUST be made before any other functions can be called.
-var initialized = false
-
-func assertInitialized() {
-	if !initialized {
-		panic("Plugin system has not been initialized")
-	}
-}
-
-var loadedPlugins []*pluginapi.Plugin
-
-// InitializeAll loads all plugins into memory and stores their symbols
+// InitializeAll loads all plugins into memory and stores their symbols.
+// A call to InitializeAll MUST be made only only once.
 func InitializeAll(libexecdir string) error {
```

Comment:
More globally the stored meta data doesn't give enough information about when to load a plugin, they are all loaded if enabled but we could potentially load some of them only at some point of the workflow.